### PR TITLE
Port coordAll function and test, also: refactor coordEach callback 

### PR DESCRIPTION
**Pull request by armantorkzaban**

### CODE REVIEW

**Comment by armantorkzaban** [comment]
can we change the signature of CoordEachCallback's   CoordinateType? currentCoord,
 to Position? currentCoord?

File: lib/src/meta.dart [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import 'geojson.dart';
 
 typedef CoordEachCallback = dynamic Function(
+  //todo: can we change this Position
```
Comment (lukas-h) [inline lib/src/meta.dart:4]: Yes, let's do it!

File: lib/src/meta.dart [hunk 2]
```diff
@@ -309,3 +310,33 @@ void featureEach(GeoJSONObject geoJSON, FeatureEachCallback callback) {
...
+///
+/// ```dart
+/// var featureColl = FeatureCollection(features:
+/// [Feature(geometry: Point(coordinates: Position(13,15)))
+/// ,Feature(geometry: LineString(coordinates: [Position(1, 2),
+/// Position(67, 50)]))]);
+///
+/// var coords = coordAll(features);
+/// //= [Position(13,15), Position(1, 2), Position(67, 50)]
+/// @Lukas: return Position or CoordinateType?
```
Comment (lukas-h) [inline lib/src/meta.dart:327]: Let's use comments for that, not directly in the source code.

`Position?` is the right type. Please refactor coordAll and coordEach to use that type!

**Review by lukas-h — CHANGES_REQUESTED** [review]

File: lib/src/meta.dart [hunk 3]
```diff
@@ -66,7 +66,7 @@ void coordEach(GeoJSONObject geoJSON, CoordEachCallback callback,
             : 0;
 
         if (geomType == GeoJSONObjectType.point) {
-          if (callback(coords as CoordinateType, coordIndex, featureIndex,
+          if (callback(coords as Position, coordIndex, featureIndex,
```
Comment (lukas-h) [inline lib/src/meta.dart:69]: is the typecast necessary anymore?

**Review by lukas-h — APPROVED** [review]

File: lib/src/meta.dart (hunk 3 above)
Comment (lukas-h) [reply lib/src/meta.dart]: Please resolve conversations, if it’s done!

**Comment by lukas-h** [comment]
@armantorkzaban when is this PR mergeable?

**Comment by lukas-h** [comment]
@armantorkzaban the merge conflict was only, because in the meantime I refactored coordEach in another branch that got merged first.

**Pull request merged**